File: py/Token.py
from enum import Enum

class TokenType(Enum):
   NUMBER = 0,
   STRING = 1,
   IDENTIFIER = 2,
   COMMA = 3,		   # ,
   DOT = 4,		   # .
   COLON = 5,		   # :
   SEMICOLON = 6,	   # ;
   LBRACKET = 7,	   # [
   RBRACKET = 8,	   # ]
   LBRACE = 9,		   # {
   RBRACE = 10,		   # }
   LPAREN = 11,		   # (
   RPAREN = 12,		   # )
   PLUS = 13,		   # +
   MINUS = 14,		   # -
   ASTERISK = 15,	   # *
   SLASH = 16,		   # /
   EQUAL = 17,		   # =
   LESS = 18,		   # <
   GREATER = 19,	   # >
   EQUAL_EQUAL = 20,   # ==
   LESS_EQUAL = 21,	   # <=
   GREATER_EQUAL = 22,  # >=
   BANG_EQUAL = 23,	   # !=
   VAR = 24,		   # var
   IF = 25,			   # if
   ELSE = 26,		   # else
   TRUE = 27,		   # true
   FALSE = 28,		   # false
   NIL = 29,		   # nil
   WHILE = 30,		   # while
   FUNCTION = 31,	   # function
   RETURN = 32,		   # return
   AND = 33,		   # and
   OR = 34,			   # or
   NOT = 35,		   # not
   STRUCT = 36,		   # struct
   UNKNOWN = 37,
   END = 38


class Token:
    type: TokenType
    literal: str
    line: int

    def __init__(self, type: TokenType, literal: str, line: int) -> None:
        self.type = type
        self.literal = literal
        self.line = line

    def Print(self):
        print("{},{}", self.literal, self.line)

File: csharp/Token.cs
namespace ComputeDuck
{
    public enum TokenType
    {
        NUMBER = 0,
        STRING,
        IDENTIFIER,
        COMMA,         // ,
        DOT,           // .
        COLON,         // :
        SEMICOLON,     // ;
        LBRACKET,      // [
        RBRACKET,      // ]
        LBRACE,        // {
        RBRACE,        // }
        LPAREN,        // (
        RPAREN,        // )
        PLUS,          // +
        MINUS,         // -
        ASTERISK,      // *
        SLASH,         // /
        EQUAL,         // =
        LESS,          // <
        GREATER,       // >
        EQUAL_EQUAL,   // ==
        LESS_EQUAL,    // <=
        GREATER_EQUAL, // >=
        BANG_EQUAL,    // !=
        VAR,           // var
        IF,            // if
        ELSE,          // else
        TRUE,          // true
        FALSE,         // false
        NIL,           // nil
        WHILE,         // while
        FUNCTION,      // function
        RETURN,        // return
        AND,           // and
        OR,            // or
        NOT,           // not
        STRUCT,        // struct
        REF,           // ref
        LAMBDA,        // lambda

        UNKNOWN,
        END
    }

  public  class Token
    {
        public Token(TokenType type, string literal, int line)
        {
            this.literal = literal;
            this.type = type;
            this.line = line;
        }

        public string Stringify()
        {
            return this.literal + "," + this.line.ToString();
        }

        public TokenType type;
        public string literal;
        int line;
    }
}